pkg/sawmill/sawmill.go

```diff
@@ -0,0 +1,87 @@
+package sawmill
+
+// If you're wondering why this package is named sawmill, I do too: https://clips.twitch.tv/PlacidOutstandingPelicanCharlieBitMe
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"cloud.google.com/go/logging"
+	"google.golang.org/api/option"
+	logpb "google.golang.org/genproto/googleapis/logging/v2"
+)
+
+// Logger holds the newly generated error logging client, so that it can be used for our own, more simple custom logging functions instead of the ones the Google provides
+type Logger struct {
+	Logger *logging.Logger // The Logger
+
+	InitError bool // If their was an error with the initialization, it won't log to the cloud
+
+	ServiceName string // The service that logging was created for
+}
+
+// InitClient creates a logging client
+func InitClient(projectID string, credentials string, environment string, serviceName string) (Logger, error) {
+	// Checks if were in the development environment, because if we are Cloud Logging is not needed
+	if environment == "DEV" {
+		return Logger{nil, false, serviceName}, nil
+	}
+
+	// The client takes the credentials in a byte array, so we do that conversion here
+	credentialsByte := []byte(credentials)
+	ctx := context.Background()
+
+	// Initializes an Google logging client
+	loggingClient, err := logging.NewClient(ctx, projectID, option.WithCredentialsJSON(credentialsByte))
```

Comment:
I guess I'm confused - I thought we agreed on not using the file here, but to use environment variables instead?

Reply:
I've set it up so that raw JSON from the ENV is passed to InitClient, and `WithCredentialsJSON' just takes that string in a byte array. This mostly for dev reasons because the ENVs in `config.yml' can't be accessed by anything but the serverless functions while in local development.

Reply:
So you will pass through a JSON file at this point? I think this should work? You tell me as I haven't tested it!

Reply:
So you pass the raw contents of the `adminSdkSecret-Dev.json` file, and yes it does work, thankfully.

Reply:
I see. Okay so what we will need to do then is actually add this JSON files to every function. Which is not _terrible_, but shall do for now!

Reply:
Although you could do it that way, what I've been doing is adding it as an ENV in `config.yaml`, so: `GCLOUD_CONFIG: {"type": "service_account", ...}`